src/containers/load-record.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Show
+ * Fetch and present a single item
+ * @flow
+ */
+
+import React from 'react'
+import { request, isValidSlug } from './request'
+import { type Record } from '../record'
+
+type Props = {
+  makeURL: (string, ?mixed) => string,
+  onError: (*) => Error,
+  onFetch: (*) => Record,
+  url: string,
+  slug: ?mixed,
+  children: ?(Result) => *
+}
+
+type State = {
+  fetching: boolean,
+  error: ?Error,
+  data: ?Record,
+  targetURL: string,
+  shouldFetch: boolean
+}
+
+export type Result = {
+  error: ?Error,
+  fetching: boolean,
+  data: *
+}
+
+const identity = (n: *) => n
+
+const makeURL = (url: string, id: *) => url + (id ? '/' + String(id) : '')
+
+export default class LoadRecord extends React.PureComponent<Props, State> {
+  lastRequest: ?XMLHttpRequest
+
+  static defaultProps = {
+    makeURL: makeURL,
+    onError: identity,
+    onFetch: identity
+  }
+
+  static getDerivedStateFromProps(props: Props, lastState: State): * {
+    let targetURL = props.makeURL(props.url, props.slug)
+
+    if (targetURL === lastState.targetURL) {
+      return null
+    }
+
+    return { targetURL, shouldFetch: isValidSlug(props.slug) }
+  }
+
+  state = {
+    data: null,
+    error: null,
+    fetching: false,
+    targetURL: '',
+    shouldFetch: false
+  }
+
+  componentDidMount() {
+    this.fetch()
+  }
+
+  componentDidUpdate() {
+    this.fetch()
+  }
+
+  fetch() {
+    if (this.lastRequest) {
+      this.lastRequest.abort()
+      this.lastRequest = null
+    }
+
+    if (this.state.shouldFetch) {
```

Comment:
`shouldFetch` implies a bit more functionality to me than something like `isValidSlug`

Reply:
Hmmm, I think this would be clearer has a method. I'll take a second pass at this upstream.

Reply:
Updated in https://github.com/vigetlabs/ars-arsenal/pull/46 to:

```javascript
class RecordFetcher extends DataFetcher {
  shouldFetch(nextURL, lastURL, props) {
    return nextURL !== lastURL && isValidSlug(props.slug)
  }
}
```